django/db/models/query.py

```diff
@@ -301,12 +301,15 @@ def __or__(self, other):
     # METHODS THAT DO DATABASE QUERIES #
     ####################################
 
-    def iterator(self):
+    def iterator(self, chunked_fetch=True):
         """
         An iterator over the results from applying this QuerySet to the
         database.
         """
-        return iter(self._iterable_class(self, chunked_fetch=True))
+        use_chunked_fetch = (
+            chunked_fetch and connections[self.db].settings_dict.get('ENABLE_SERVER_SIDE_CURSORS', True)
+        )
+        return iter(self._iterable_class(self, chunked_fetch=use_chunked_fetch))
```

Comment:
I do like the solution where you can opt-out per query, or force opt-out for all queries on given connection. This way Django and library code can use cursors without a risk of being incompatible with pgbouncer.